datastore_orm/model.py

```diff
@@ -730,6 +740,32 @@ def get_multi(self, keys, missing=None, deferred=None,
         """
         if not keys:
             return []
+        get_multi_partial = partial(self.get_single, missing=missing, deferred=deferred, transaction=transaction,
+                                    eventual=eventual, model_type=model_type)
+        ordering = dict()
+        with ThreadPoolExecutor(max_workers=len(keys)) as executor:
```

Comment:
you need to cap max_workers at a level. if you have too many workers it will actually slow down.